src/components/dynamic/form/Admin.jsx

```diff
@@ -19,27 +21,60 @@ const Admin = () => {
   });
 
   const handleSubmit = (setLoading, setState) => {
-    api({
+    apiCallWithStatusCode({
       method: "POST",
       url: "/api/dashboard/admins",
       body: admin,
     })
-      .then(() => toast(`✅ Submitted successfully!`))
-      .catch(() => toast(`❌ Internal Server Error`))
+      .then(({ responseBody, statusCode }) => {
+        if (statusCode == 200) {
+          toast(`✅ Submitted successfully!`);
+        } else if (statusCode == 400) {
+          toast(`❌ Invalid input`);
+        } else {
+          setErrorObj({
+            code: statusCode,
+            name: "Internal Server Error",
+            message: "Please reach out the dev team with the below message",
+            dev: responseBody.message,
+          });
+        }
+      })
+      .catch((error) => {
+        if (!errorObj) {
+          setErrorObj({
```

Comment:
this should be thrown instead, checkout the protected page, and how it implements throwing errors

Reply:
the onSubmit is swallowing the error being thrown, it blocks it from the error message appearing for some reason
https://github.com/redux-form/redux-form/issues/1859